### PR TITLE
feat: add api pentest

### PR DESCRIPTION
**Pull request by 0xlildoudou**
Hi Team,

I recently did an API pentest and suggest adding a few notes to “The Hackers Recipes”.

Thanks for your work and the redesign of the site.

### CODE REVIEW

File: docs/src/web/inputs/api.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+Older API versions might be more susceptible to attacks. Always check for and test against multiple API versions.
+
+Example:
+```
+https://taget.com/api/v2/users/1 --> 401
+https://taget.com/api/v1/users/1 --> 200
+```
+
+## Sources
```
Comment (AzeTIIx) [inline docs/src/web/inputs/api.md:95]: Refactor "Sources" to "Ressources"

File: docs/src/web/inputs/api.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+:::
+
+### Parameter Tampering
+
+Experiment with adding or replacing parameters in requests to access unauthorized data or functionalities.
+
+Example:
+```
+https://taget.com/api/users/1 --> 401
```
Comment (AzeTIIx) [inline docs/src/web/inputs/api.md:80]: "taget" -> "target"

File: docs/src/web/inputs/api.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+## Theory
+
+The API pentesting methodology begins with reconnaissance, where information is gathered about the API, including its endpoints, parameters, and authentication methods. Next, testers assess authentication and authorization to ensure proper access control and attempt to bypass them. Input validation is then tested by sending crafted payloads to identify vulnerabilities like injection attacks. Business logic testing follows, ensuring the API handles data and functions correctly without unintended consequences. Rate limiting and denial-of-service (DoS) protections are evaluated to prevent excessive requests. Finally, a detailed report is created, documenting findings, vulnerabilities, and recommendations for remediation.
+
+## Practice 
+
+### API Types
+* SOAP/XML Web Services : SOAP (Simple Object Access Protocol) is an XML-based protocol for exchanging structured information in the implementation of web services. 
+* REST APIs (JSON) : The specificity of a REST API lies in its statelessness, use of standard HTTP methods (GET, POST, PUT, DELETE), and resource-based structure with responses typically formatted in JSON or XML.
```
Comment (AzeTIIx) [inline docs/src/web/inputs/api.md:15]: Add backticks to HTTP methods GET -> `GET`

**Comment by 0xlildoudou** [comment]
Hi,

I have fix your comment, thanks :)

**Comment by AzeTIIx** [comment]
One last thing, when I said "refactor taget -> target", I meant it for all the occurences of "taget" 
I didn't make it clear enough, my bad 😅

File: docs/src/web/inputs/api.md [hunk 4]
```diff
@@ -55,34 +55,34 @@ Wordlist for discover API endpoints
...
-### Parameter Tampering
+### Parameter tampering
 
 Experiment with adding or replacing parameters in requests to access unauthorized data or functionalities.
 
 Example:
 ```
-https://taget.com/api/users/1 --> 401
+https://target.com/api/users/1 --> 401
 https://taget.com/api?users=1 --> 200
```
Comment (AzeTIIx) [inline docs/src/web/inputs/api.md:81]: taget -> target

File: docs/src/web/inputs/api.md [hunk 5]
```diff
@@ -55,34 +55,34 @@ Wordlist for discover API endpoints
...
 https://taget.com/api/users?1 --> 200
```
Comment (AzeTIIx) [inline docs/src/web/inputs/api.md:82]: taget -> target

File: docs/src/web/inputs/api.md [hunk 6]
```diff
@@ -92,6 +92,6 @@ https://taget.com/api/v2/users/1 --> 401
 https://taget.com/api/v1/users/1 --> 200
```
Comment (AzeTIIx) [inline docs/src/web/inputs/api.md:92]: taget -> target

**Review by AzeTIIx — CHANGES_REQUESTED** [review]

**Comment by 0xlildoudou** [comment]
it's fixed
I'm so fucking stupid...

**Comment by 0xlildoudou** [comment]
Hi,

All is done, can you merge my MR or do you need something else?

**Review by AzeTIIx — APPROVED** [review]

**Pull request merged**